SwiftGenStrings/CharacterIterator.swift

```diff
@@ -47,4 +47,21 @@ class CharacterIterator {
         return characters[index - 1]
     }
 
+    func advance(_ count: Int) {
+        if index + count < characters.count {
+            index += count
+        } else {
+            index = characters.count - 1
+        }
+    }
+    
+    func charactersMatch(_ string: String) -> Bool {
```

Comment:
nit: what about `starts(with:)`? That would make it more similar to `IndexingIterator`